storage/storage_image.go

```diff
@@ -682,6 +810,31 @@ func (s *storageImageDestination) Commit(ctx context.Context, unparsedToplevel t
 			lastLayer = layer.ID
 			continue
 		}
+
+		if diffOutput := s.diffOutputs[blob.Digest]; diffOutput != nil {
+			layer, err := s.imageRef.transport.store.CreateLayer(id, lastLayer, nil, "", false, nil)
+			if err != nil {
+				return err
+			}
+
+			// FIXME: what to do with the uncompressed digest?
+			diffOutput.UncompressedDigest = blob.Digest
+
+			if err := s.imageRef.transport.store.ApplyDiffFromStagingDirectory(layer.ID, diffOutput.Target, diffOutput, nil); err != nil {
```

Comment:
Is the two-stage process really _required_ for this?

We’d already like for the image pull process to happen concurrently for layers as they arrive (pull in any order, and create layers in order as soon as possible, instead of currently just staging everything and then doing a lot of work in `Commit`). If that’s the more natural approach for this feature, it would probably make sense to return to that work and get it working, and have a simpler implementation and a more straightforward c/storage API.